src/vaadin-date-time-picker.html

```diff
@@ -258,7 +266,25 @@
             this.value = formattedValue;
           }
 
-          this.$.customField.value = formattedValue;
+          // Setting the customField.value below triggers validation of the date picker and time picker.
+          // If the inputs are slotted (e.g. when using the Java API) and have an initial value this can
+          // happen before date picker ready() which triggers an error when date picker is trying to read
+          // `this.$.input` (as a result of value change triggered by setting custom field value).
+          // Workaround the problem by overriding customField.validate while setting the initial value.
+          if (!this.$.customField.inputs[0].$) {
+            const validate = this.$.customField.validate;
+            /* istanbul ignore next */
+            this.$.customField.validate = () => true;
+
+            this.$.customField.value = this.value;
+
+            this.$.customField.validate = validate;
+            setTimeout(() => {
+              this.$.customField.validate();
+            });
+          } else {
+            this.$.customField.value = this.value;
+          }
```

Comment:
This is kind of ugly, but I couldn't think of any nicer way to solve the problem right now. At least this seems to work and it would be good to get merged and released as alpha2 soon because it's blocking `vaadin-date-time-picker-flow` MVP.

Maybe this could be avoided if custom field would handle this situation automatically and not calling the validation of the inputs too early, or maybe date-picker should handle it by not trying to read the `$.input` if validate is called too early. I'm not sure should it be allowed to trigger validation before `ready()` or not.

Reply:
Just create a ticket to find a proper solution and refactoring this so we don't forget it.

Reply:
Will create an issue right after merging this so I can link to the code in master branch.

Reply:
Issue here #13